### ISSUE TITLE
Sensibo: Fan controls now listed in alphabetical order

### ISSUE THREAD

**Issue by Talismanian** [nervetattoo/simple-thermostat#397]
**Describe the bug**
Basically, fan controls aren't in the order listed in config anymore, but in alphabetical order. This has occurred with HA 2023.8.0, supervisor 2023.07.1, OS 10.4, frontend 20230802.0.

**To Reproduce**
Set up your thermostat card with the fan control speeds listed for your climate entity. If they're labelled alphabetically, you will see them ordered in the same way, regardless of where you position them in the config.

**Expected behavior**
In previous versions (HA 2023.7 and earlier), the card displayed the fan speeds in order as listed in the config.

```yaml
entity: climate.downstairs
type: custom:simple-thermostat
layout:
  step: row
  mode:
    headings: false
    names: false
step_size: '1'
decimals: '0'
control:
  hvac: true
  fan:
    _hide_when_off: true
    quiet: false
    strong: false
    low:
      name: Low
      icon: mdi:fan-speed-1
    medium:
      name: Medium
      icon: mdi:fan-speed-2
    high:
      name: High
      icon: mdi:fan-speed-3
    auto:
      name: Auto
      icon: mdi:fan-auto
```

**Screenshots**
![image](https://github.com/nervetattoo/simple-thermostat/assets/28279957/29ead6eb-d4c6-458a-b737-c549c16a3a11)

(I've written in the names of the fan modes, A for auto, H for high, L for low, M for medium. Notice how they're now in alphabetical order.)

**Browser**
 - OS: Win, iOS
 - Browser: Firefox, Mobile app
 - Simple Thermostat version from browser console: 2.5.0
 - Home Assistant version: 2023.8.0

**Comment by gitrummy** [comment]
+1

**Comment by TTLucian** [comment]
Same here

**Comment by Talismanian** [comment]
Hi everyone, I wanted to find out what integration was serving your climate entities. In my case, it's **Sensibo**. It seems this bug might not be universal across all integrations.

**Comment by gitrummy** [comment]
> Hi everyone, I wanted to find out what integration was serving your climate entities. In my case, it's **Sensibo**. It seems this bug might not be universal across all integrations.

Sensibo for me :)

**Comment by TTLucian** [comment]
Sensibo

**Comment by Talismanian** [comment]
I've noticed the issue is related to the order in which fan speeds are enumerated in the entity's ```fan_modes``` attribute.

I can temporarily change the order of the fan modes by setting the attributes in Developer Tools > States. But the order will revert back after a moment when HA refreshes the entity from Sensibo.

I'm waiting to hear back from a friend with an older version of the Sensibo integration installed to double-check if this behaviour was introduced in the latest update.

**Comment by gitrummy** [comment]
> I've noticed the issue is related to the order in which fan speeds are enumerated in the entity's `fan_modes` attribute.
> 
> I can temporarily change the order of the fan modes by setting the attributes in Developer Tools > States. But the order will revert back after a moment when HA refreshes the entity from Sensibo.
> 
> I'm waiting to hear back from a friend with an older version of the Sensibo integration installed to double-check if this behaviour was introduced in the latest update.

I've got heaps of backups if you need someone to go back and test anything

**Comment by Talismanian** [comment]
FYI I've tested on an older version of HA pre-2023.8 and the behaviour is normal. I've logged an issue on HA Core to see if there's any way of fixing this. But the ultimate resolution would be for Simple Thermostat to arrange fan speeds by their order listed in config.

**Comment by Talismanian** [comment]
FYI a [PR has been merged in HA core for changes to pysensibo](https://github.com/home-assistant/core/pull/100245). It claims to directly address this issue around fan modes being enumerated alphabetically. Let's wait and see how this goes once it's pushed out to the release.

**Comment by Talismanian** [comment]
Issue has been resolved in HA 2023.10
